File: src/main/java/com/indiepost/service/AnalyticsService.java
package com.indiepost.service;

import com.indiepost.dto.analytics.*;

/**
 * Created by jake on 17. 4. 13.
 */
public interface AnalyticsService {

    OverviewStats getOverviewStats(PeriodDto periodDto);

    RecentAndOldPostStats getRecentAndOldPostStats(PeriodDto periodDto);

    PostStatsDto getPostStats(PeriodDto periodDto);

    PostStatsDto getCachedPostStats();

    TopStats getTopStats(PeriodDto periodDto);
}

File: src/main/kotlin/com/indiepost/service/AnalyticsService.kt
package com.indiepost.service

import com.indiepost.dto.analytics.*

/**
 * Created by jake on 17. 4. 13.
 */
interface AnalyticsService {

    fun getCachedPostStats(): PostStatsDto?

    fun getOverviewStats(periodDto: PeriodDto): OverviewStats

    fun getRecentAndOldPostStats(periodDto: PeriodDto): RecentAndOldPostStats

    fun getPostStats(periodDto: PeriodDto): PostStatsDto

    fun getTopStats(periodDto: PeriodDto): TopStats
}
